concourse/model/traits/image_scan.py

```diff
@@ -84,6 +84,26 @@
         doc='CVSS version used to evaluate the severity of vulnerabilities',
         type=CVSSVersion,
     ),
+    AttributeSpec.optional(
+        name='allowed_licenses',
+        default=None,
+        doc=(
+            'A list of regular expressions. If configured, licenses detected by protecode '
+            'that do not match at least one of regular expressions will result in a report mail '
+            'being sent. If not configured (i.e. `None`), **all** licenses will be accepted.'
+        ),
+        type=list,
+    ),
+    AttributeSpec.optional(
+        name='prohibited_licenses',
+        default=[],
```

Comment:
for consistency reasons, use either `None`, or an empty sequence in both cases. for avoiding unexpected side-effects, use immutable sequence (aka tuple)

Reply:
also: what happens if both allowed and prohibited licenses are defined (should be documented)?

I would _expect_ the following behavour:

1. allowed licences are defined, but prohibited licences are not defined
-> any license that is not explicitly allowed, is assumed to be prohibited

2. prohibited licenses are defined, but allowed licences are not
-> any license that is not explicitly prohibited is allowed

3. both allowed and prohibited licences are defined
-> seems difficult to define a reasonable semantics
we might in this case differentiate:
- allowed licenses are allowed (no-brainer)
- prohibited licenses are prohibilted (also no-brainer)
- all other licences are "unclassified"